zanata-war/src/main/java/org/zanata/adapter/GettextAdapter.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2015, Red Hat, Inc. and individual contributors as indicated by the
+ * @author tags. See the copyright.txt file in the distribution for a full
+ * listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This software is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
+ * site: http://www.fsf.org.
+ */
+
+package org.zanata.adapter;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang.CharSet;
+import org.apache.commons.lang.StringUtils;
+import org.xml.sax.InputSource;
+import org.zanata.adapter.po.PoReader2;
+import org.zanata.adapter.po.PoWriter2;
+import org.zanata.common.LocaleId;
+import org.zanata.common.io.FileDetails;
+import org.zanata.dao.DocumentDAO;
+import org.zanata.exception.FileFormatAdapterException;
+import org.zanata.file.GlobalDocumentId;
+import org.zanata.model.HDocument;
+import org.zanata.rest.StringSet;
+import org.zanata.rest.dto.resource.Resource;
+import org.zanata.rest.dto.resource.TextFlowTarget;
+import org.zanata.rest.dto.resource.TranslationsResource;
+import org.zanata.rest.service.ResourceUtils;
+import org.zanata.util.FileUtil;
+import org.zanata.util.ServiceLocator;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Optional;
+import com.google.common.collect.Sets;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Adapter for reading and write {@link org.zanata.common.DocumentType#GETTEXT_PORTABLE_OBJECT}
```

Comment:
- [ ] .(grammar) "... for reading and writing ..." or "... to read and write ..."